### PR TITLE
Add Japanese language support & translate some docs into Japanese

### PR DESCRIPTION
**Pull request by morinokami**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content
- Changes to Starlight code

#### Description

- Add Japanese language support.
- Translate some pages of the docs into Japanese.

### CODE REVIEW

**Comment by morinokami** [comment]
@delucis While this PR includes changes to the Starlight code, the majority of the changes are translations. Should I add a changeset in this case?

**Comment by delucis** [comment]
Thanks for this @morinokami!

Yes, we should add a patch changeset so that the next release notes say that Japanese support was added 🥳

You can add one by running `pnpm changeset add` in the repo or clicking the “add a changeset” link in the bot message above. Let me know if you have any questions.

We should also update this line in the i18n docs to mention Japanese support: https://github.com/withastro/starlight/blob/de24b54971577912979a3fb67570f4c95efe27a6/docs/src/content/docs/guides/i18n.md?plain=1#L113

File: docs/src/content/docs/ja/index.mdx [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+title: ようこそ、世界
+description: Starlight helps you build beautiful, high-performance documentation websites with Astro. Starlightは、Astroを使って美しく高性能なドキュメントサイトを作るためのテーマです。
```
Comment (delucis) [inline docs/src/content/docs/ja/index.mdx:3]: Looks like the English got left in here:

```suggestion
description: Starlightは、Astroを使って美しく高性能なドキュメントサイトを作るためのテーマです。
```

**Review by delucis — APPROVED** [review]
Thanks again @morinokami. I made those changes so I can release this tonight — will also make one more tweak.

**Comment by morinokami** [comment]
@delucis Ah, it was the middle of the night in Japan, so I had to stop working on it😅 Thank you for taking over and completing the rest!

**Pull request merged**